### ISSUE TITLE
Problem synching with version 3.1.1

### ISSUE THREAD

**Issue by tinxx** [Kozea/Radicale#1209]
That was a nasty shock when I realized this morning that all my contacts, appointments and tasks on my phone were gone.

I upgraded to 3.1.1 yesterday and there seems to be something wrong with synching now. Both DAVx and Thunderbird seem to have problems communicating with Radicale ver 3.1.1.

I quickly realized that all the files under `data` were still intact and added the [proposed commit hook](https://radicale.org/v3.html#versioning-with-git) that introduces versioning for all content and configs.

I "solved" the problem by pinning Radicale to version 3.1.0 for the time being.

Please see into the problem and feel free to hit me for further details.

Cheers,
tinxx

**Comment by Unrud** [comment]
Are you using a reverse proxy or an external WSGI server?
Can you provide a log with the `--debug` argument?

**Comment by tinxx** [comment]
Hello @Unrud,
thanks for the quick response!

I'm using a NGINX reverse proxy. Worked splendidly before the 3.1.1 upgrade.

I cannot debug right now because a dist upgrade is running on the machine.

**Comment by Unrud** [comment]
> I'm using a NGINX reverse proxy.

Can you provide the configuration? Radicale 3.1.1 doesn't sanitizes the SCRIPT_NAME. I strongly suspect that's the issue here.

**Comment by tinxx** [comment]
Thanks for the hint, I'm gonna check out [the fine manual](https://radicale.org/v3.html#manage-user-accounts-with-the-reverse-proxy) when the system upgrade has completed.

**Comment by tinxx** [comment]
Okay, I think I solved the problem.

In the example from the manual the config of the problematic header looks like this:

```
location /radicale/ {
    proxy_pass           http://localhost:5232/;
    proxy_set_header     X-Script-Name /radicale;
}
```

Now in my config I adapted it and set

```
    proxy_set_header X-Script-Name /;
```

That does not work.  
In the debug log I could see that Radicale 3.1.0 sanitized it to "".

In my config I changed it to the following, which does not send the problematic header at all. So removing it would probably also work.

```
    proxy_set_header X-Script-Name "";
```

Thanks again, @Unrud, for the help!

Cheers,
tinxx
